File: examples/canonical-api/ruby-hanami-example/app/router.rb
require "hanami/router"

require_relative "actions/health/show"
require_relative "actions/reports/show"
require_relative "actions/report_cards/show"
require_relative "actions/charts/show"

module RubyHanamiExample
  App = Hanami::Router.new do
    get "/healthz", to: Actions::Health::Show.new
    get "/api/reports/:tenant_id", to: Actions::Reports::Show.new
    get "/fragments/report-card/:tenant_id", to: Actions::ReportCards::Show.new
    get "/data/chart/:metric", to: Actions::Charts::Show.new
  end
end

File: examples/canonical-api/phoenix-example/lib/example_web/router.ex
defmodule ExampleWeb.Router do
  use Phoenix.Router

  pipeline :api do
    plug :accepts, ["json"]
  end

  pipeline :fragment do
    plug :accepts, ["html"]
  end

  scope "/", ExampleWeb do
    pipe_through :api

    get "/healthz", HealthController, :show
    get "/api/reports/:tenant_id", ReportController, :index
    get "/data/chart/:metric", ChartController, :show
  end

  scope "/fragments", ExampleWeb do
    pipe_through :fragment

    get "/report-card/:tenant_id", FragmentController, :show
  end
end
